### ISSUE TITLE
Random port for Alice's listen address in the tests

### ISSUE THREAD

**Issue by da-kami** [comit-network/xmr-btc-swap#151]
## Problem

Currently we use `get_port` to get a free port when starting tests.
This seems to cause trouble when running tests in parallel. Generally, it is unnecessary to depend on  another library for this, when we can automatically get a random free port assigned by the operating system.

## Goal

Let the OS assign a port by passing `0` to the multi-addr. 
We need to wait for the `NewListenAddr` event then but we need to do that anyway to make sure the Swarm is ready.

**Comment by thomaseizinger** [comment]
> when libp2p actually has the capability to assign a random port.

Nitpick: This is not functionality of libp2p but of your OS kernel. Passing 0 as the port to listen on will give you a random port that is available.

**Comment by thomaseizinger** [comment]
Already done.
